src/json.h

```diff
@@ -0,0 +1,52 @@
+/* ****************************************************************************
+ *
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ *
+ *
+ * This file is part of Microsoft R Host.
+ *
+ * Microsoft R Host is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Microsoft R Host is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Microsoft R Host.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * ***************************************************************************/
+
+#pragma once
+#include "stdafx.h"
+#include "util.h"
+#include "Rapi.h"
+
+namespace rhost {
+    namespace json {
+        // Produces JSON from an R object according to the following mappings:
+        //
+        // NULL, NA, empty vector -> null
+        // TRUE -> true
+        // FALSE -> false
+        // Vector of a single non-NA integer or double -> numeric literal
+        // Vector of a single non-NA string -> string literal
+        // List with all elements unnamed -> array (recursively)
+        // List with all elements named, or environment -> object (recursively)
+        //
+        // If any element of a list or environment is NA, that element is skipped.
```

Comment:
Can you explain why we have to skip NA elements?

While that may not be a problem for data that we create ourselves (we can avoid putting NA elements in the first place), if we want to serialize a list that is created by the user or R library code, we'll get unexpected results.